languages/haskell/docs/implementing-a-concept-exercise.md

```diff
@@ -1,5 +1,54 @@
-# How to implement an Haskell concept exercise
+# How to implement a Haskell concept exercise
 
-TODO: describe how to implement a concept exercise for the Haskell track. For inspiration, check out the [C# version of this file][csharp-implementing].
+This document describes the steps required to implement a concept exercise for the Haskell track. As this document is generic, the following placeholders are used:
 
-[csharp-implementing]: ../../csharp/docs/implementing-a-concept-exercise.md
+- `<SLUG>`: the name of the exercise in kebab-case (e.g. `anonymous-methods`).
+- `<NAME>`: the name of the exercise in PascalCase (e.g. `AnonymousMethods`).
+
+Before implementing the exercise, please make sure you have a good understanding of what the exercise should be teaching (and what not). This information can be found in the exercise's GitHub issue. Concept exercises should be modelled after the CIS 194 Haskell course at Penn University, either the [Spring 2013][spring-2013] or [Fall 2016][fall-2016] version.
+
+To implement a concept exercise, the following files need to be created:
+
+```
+languages
+└── Haskell
+    └── exercises
+        └── concept
+            └── <SLUG>
+                ├── .docs
+                |   ├── instructions.md
+                |   ├── introduction.md
+                |   ├── hints.md
+                |   └── after.md (optional)
+                ├── .meta
+                |   ├── config.json (omit for now)
+                |   └── <SLUG>.hs (example solution)
```

Comment:
I would suggest using `Example.hs` or `<Name>.Example.hs`, as the default CLI settings will make sure files that have "example" (in any casing) in their name will never be downloaded.

```suggestion
                |   └── Example.hs (example solution)
```

Reply:
From @sshine's comment:

> The current Haskell track repo names the example solution file the same as the exercise, but places it in a separate directory. This means that the files examples/success-{foo, bar}/{package.yaml, src/<NAME>.hs} exist: Multiple sample solutions can exist, and the track repo's CI suite can test the examples by copying files. This pull request appears to propose that we instead name the example solution Example.hs. The drawback associated with naming it this is that we cannot re-use the test suite on the example solution, since the test suite contains a line like import <NAME> (...) which must be replaced with import Example (...) during CI.

Reply:
Ah my bad. Sorry!